research-hub-web/src/app/app.component.html

```diff
@@ -1,301 +1,266 @@
+
+                  <!-- Logo/Home -->
+                  <a 
+                    fxFlex.lt-md 
+                    class="hub-logo"
+                    routerLink="/home">
+                    <img [src]="'assets/images/hub-logo.svg'" alt="logo">
+                  </a>
+
+                  <!-- Space Divider -->
+                  <span fxHide.lt-md style="flex: 1 1 auto;"></span>
+
+                  <!-- Search Icon (Mobile)-->
+                  <button role="button" aria-label="search" *ngIf="!onSearchPage" (click)="clearSearchText()"
+                    (keydown.enter)="clearSearchText()" mat-icon-button
+                    fxHide.gt-sm routerLink="/search" matTooltip="Search">
+                    <mat-icon>search</mat-icon>
+                  </button>
+
+                  <!-- Search Button -->
+                  <button role="button" aria-label="search" *ngIf="!onSearchPage" mat-button 
+                  fxHide.lt-md routerLink="/search" (click)="clearSearchText()" 
+                  (keydown.enter)="clearSearchText()">Search</button>
+
+                  <!-- Research Categories Button -->
+                  <button role="button" aria-label="categories" *ngIf="!onSearchPage" 
+                    mat-button fxHide.lt-md class="button-menu" [matMenuTriggerFor]="browseMenu">Research Categories</button>
+
+                  <mat-menu #browseMenu="matMenu" [overlapTrigger]="false">
+                    <div *ngIf="allCategories$ | async as allCategories">
+                      <button role="button" aria-label="category" mat-menu-item *ngFor="let item of allCategories.items"
+                        [style.background]="showBanner ? '' : '#0294a5'" [routerLink]="['/search']"
+                        (click)="searchBarService.setCategory([item.sys.id]); sidenav.close()">{{item.name}}</button>
+                    </div>
+                  </mat-menu>
+
+                  <!-- Research Activities Button -->
+                  <button role="button" aria-label="activities" *ngIf="!onSearchPage" class="button-menu" 
+                    fxHide.lt-md mat-button [matMenuTriggerFor]="researchActivityMenu">Research Activities</button>
+
+                    <mat-menu #researchActivityMenu="matMenu" [overlapTrigger]="false">
+                    <div *ngIf="allStages$ | async as allStages">
+                      <button role="button" aria-label="activity" mat-menu-item *ngFor="let item of allStages.items"
+                        [style.background]="showBanner ? '' : '#0294a5'" [routerLink]="['/search']"
+                        (click)="searchBarService.setStage([item.sys.id]); sidenav.close()">{{ item.name }}</button>
+                    </div>
+                  </mat-menu>
+
+                  <!-- Sign In Button -->
+                  <button role="button" aria-label="sign-in" *ngIf="!authenticated" fxHide.lt-md
+                    mat-button (click)="loginService.doLogin()" (keydown.enter)="loginService.doLogin()">Sign In</button>
```

Comment:
dologin should be passed the current url. Maybe cleaner to have a login function in the app.component.ts e.g. 
async login() {
    await this.loginService.doLogin(this.router.url);
}

Reply:
yep, i.e. this.router.url. This is the redirect url to take the user back to after they have done SSO. If no url is passed, it will redirect back to home page. Which isn't ideal if they clicked sign in from a different page.

Reply:
redirecting properly now 👍